src/ansiblelint/rules/jinja.py

```diff
@@ -131,58 +344,175 @@ def test_var_spacing(
+                "spacing",
+                id="19",
+            ),
+            pytest.param(
+                "{{ {'a': {'b': 'x', 'c': y}} }}",
+                "{{ {'a': {'b': 'x', 'c': y}} }}",
+                "spacing",
+                id="20",
+            ),
+            pytest.param(
+                "2*(1+(3-1)) is {{ 2 * {{ 1 + {{ 3 - 1 }}}} }}",
+                "2*(1+(3-1)) is {{ 2 * {{1 + {{3 - 1}}}} }}",
+                "spacing",
+                id="21",
+            ),
+            pytest.param(
+                '{{ "absent"\nif (v is version("2.8.0", ">=")\nelse "present" }}',
+                "",
+                "invalid",
+                id="22",
+            ),
+            pytest.param(
+                '{{lookup("x",y+"/foo/"+z+".txt")}}',
+                '{{ lookup("x", y + "/foo/" + z + ".txt") }}',
+                "spacing",
+                id="23",
+            ),
+            pytest.param(
+                "{{ x | map(attribute='value') }}",
+                "{{ x | map(attribute='value') }}",
+                "spacing",
+                id="24",
+            ),
+            pytest.param(
+                "{{ r(a= 1,b= True,c= 0.0,d= '') }}",
+                "{{ r(a=1, b=True, c=0.0, d='') }}",
+                "spacing",
+                id="25",
+            ),
+            pytest.param("{{ r(1,[]) }}", "{{ r(1, []) }}", "spacing", id="26"),
+            pytest.param(
+                "{{ lookup([ddd ]) }}", "{{ lookup([ddd]) }}", "spacing", id="27"
+            ),
+            pytest.param(
+                "{{ [ x ] if x is string else x }}",
+                "{{ [x] if x is string else x }}",
+                "spacing",
+                id="28",
+            ),
+            pytest.param(
+                # "{% if a|int <= 8 -%} iptables {%- else -%} iptables-nft {%- endif %}",
+                "{% if a|int <= 8 -%} iptables {%- else -%} iptables-nft {%- endif %}",
+                "{% if a | int <= 8 -%} iptables {%- else -%} iptables-nft {%- endif %}",
+                "spacing",
+                id="29",
+            ),
+            pytest.param(
+                # "- 2" -> "-2", minus does not get separated when there is no left side
+                "{{ - 2 }}",
+                "{{ -2 }}",
+                "spacing",
+                id="30",
+            ),
+            pytest.param(
+                # "-2" -> "-2", minus does get an undesired spacing
+                "{{ -2 }}",
+                "{{ -2 }}",
+                "spacing",
+                id="31",
+            ),
+            pytest.param(
+                # array ranges do not have space added
+                "{{ foo[2:4] }}",
+                "{{ foo[2:4] }}",
+                "spacing",
+                id="32",
+            ),
+            pytest.param(
+                # array ranges have the extra space removed
+                "{{ foo[2: 4] }}",
```

Comment:
In the AST, this is a Slice node.

You will need to track the state to successfully identify this. If within `[` skip the spaces. Maybe have a stack var, and every time you enter a left operator, put it on the stack (a list), then pop it once you hit the closing token. So, if within `{}`, `:` gets a space after. If within `[]` no spaces around `:`.

Reply:
No AST/black for the moment, likely to open another can of worms. Time will tell if we need them or not.